In [2]:
## ler as propriedades title e content do arquivo trn.json para o finetuning do modelo
import json
import os

def solve_relative_path(file_path: str) -> str:
    project_root: str = os.path.dirname(os.getcwd())
    data_path: str = os.path.abspath(os.path.join(project_root, file_path))
    return data_path

def read_json_file(file_path: str) -> list[tuple[str, str]]:
    training_data = []

    with open(file_path, "r") as file:
        buffer: str = "" 
        for line in file:
            buffer += line.strip()
            try:
                item: dict = json.loads(buffer)
                buffer = ""
                title: str = item.get("title", "")
                content: str = item.get("content", "")
                training_data.append((title, content))
            except json.JSONDecodeError:
                continue
    return training_data

if __name__ == "__main__":
    file_path = solve_relative_path("datasets/trn.json")
    training_data = read_json_file(file_path)
    for title, content in training_data[:10]:
        print(f"Title: {title}\nContent: {content}\n")

Title: Girls Ballet Tutu Neon Pink
Content: High quality 3 layer ballet tutu. 12 inches in length

Title: Adult Ballet Tutu Yellow
Content: 

Title: The Way Things Work: An Illustrated Encyclopedia of Technology
Content: 

Title: Mog's Kittens
Content: Judith Kerr&#8217;s best&#8211;selling adventures of that endearing (and exasperating) cat Mog have entertained children for more than 30 years. Now, even infants and toddlers can enjoy meeting this loveable feline. These sturdy little board books&#8212;with their bright, simple pictures, easy text, and hand&#8211;friendly formats&#8212;are just the thing to delight the very young. Ages 6 months&#8211;2 years.

Title: Misty of Chincoteague
Content: 

Title: Hilda Boswell's treasury of children's stories: A new anthology of stories for the young
Content: 

Title: The Simple Truths of Service: Inspired by Johnny the Bagger
Content: 

Title: Girls Ballet Tutu Neon Blue
Content: Dance tutu for girls ages 2-8 years. Perfect for dance practice

- Alguns valores de content estão vazios
- Alguns valores de content possuem hash de caracteres especiais. Exemplo: Judith Kerr&#8217;s best&#8211;selling
- Alguns valores de content estão da forma esperada